reqresp/README.md

```diff
@@ -0,0 +1,73 @@
+
+* Query poll - Caller can run queries polling for response (the slow/inefficient way)
+* Response activity - Caller can provide activity information on where to send the response to (the more robust way)
+
+The workflow in this specific example accepts requests to uppercase a string via signal and then provides the response
+via either approach above.
+
+### Running
+
+Follow the below steps to run this sample:
+
+1) You need a Temporal service running. See details in README.md.
+
+2) Run the following command in the background or in another terminal to run the worker:
+
+    go run ./reqresp/worker
+
+3) Run the following command to start the workflow:
+
+    go run ./reqresp/starter
+
+4) Run the following command to uppercase some strings via the slower query-response approach:
+
+    go run ./reqresp/request foo bar
+
+Among other log lines, this will output the following after a few seconds:
+
+    Uppercased - from foo to FOO
+    Uppercased - from bar to BAR
+
+5) Run the following commands to uppercase some strings via the faster activity-response approach:
+
+    go run ./reqresp/request -activity-based foo bar
+
+This will output the same thing, but much faster.
+
+### Explanation of response handling
+
+When a query is executed in Temporal, it must go to the worker, invoke the query handler (potentially replaying if the
+workflow is not already present/cached), and return the query back. Therefore, in order to do a query-based
+request/response, one must signal the workflow and continuously query the workflow for the response. The query can't be
+executed too frequently as this affects server and worker resources. This is a method that, while simple, is much
+slower.
+
+Alternatively, a Temporal workflow can execute activities on different task queues. Therefore, a "response" task queue
+can be setup to handle activities executed from the other workflow. This means the response can be pushed. By providing
+which activity and task queue the response should be sent to inside the request, the local requester can just be
+notified when the local activity is executed.
+
+In this particular example, we have abstracted both concepts out into a `Requester`. This requester can be reused
+although the sample just shows it used ephemerally as part of the `request` CLI.
+
+### Explanation of continue-as-new
+
+Workflows cannot have infinitely-sized history and when the event count grows too large, `ContinueAsNew` can be returned
+to start a new one atomically. However, in order not to lose any data, signals must be drained and any other futures
+that need to be reacted to must be completed first. This means there must be a period where there are no signals to
+drain and no futures to wait on. If signals come in faster than processed or futures wait so long there is no idle
+period, `ContinueAsNew` will never happen and the history will grow unchecked until the extreme limit is reached at the
+server causing workflow termination.
```

Comment:
I don't believe this is the case. I think there is a hard limit on a number of signals a workflow can receive. After that the new signals are rejected, but the workflow is not terminated.

Reply:
I just assumed that from https://docs.temporal.io/docs/go/workflows/#large-event-histories. I was not aware of a separate signal limit. I will remove this detail anyways because the reader should not be concerned with what happens in this extreme case we're trying to avoid anyways.

Reply:
Updated to avoid details about what happens on uncapped signal count